src/test/java/mate/academy/dao/RoleDaoTest.java

```diff
@@ -0,0 +1,47 @@
+package mate.academy.dao;
+
+import java.util.Optional;
+import mate.academy.dao.impl.RoleDaoImpl;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.model.Role;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RoleDaoTest extends AbstractTest {
+    private static final String ROLE_NAME = "USER";
+    private Role role;
+    private RoleDao roleDao;
+
+    @Override
+    protected Class<?>[] entities() {
+        return new Class[] {Role.class};
+    }
+
+    @BeforeEach
+    void setUp() {
+        role = new Role();
+        role.setRoleName(Role.RoleName.USER);
+        roleDao = new RoleDaoImpl(getSessionFactory());
+    }
+
+    @Test
+    void save_ok() {
+        Role actual = roleDao.save(role);
+        Assertions.assertNotNull(actual);
+        Assertions.assertEquals(1L, actual.getId());
+    }
+
+    @Test
+    void getRoleByName_ok() {
+        roleDao.save(role);
+        Optional<Role> actual = roleDao.getRoleByName(ROLE_NAME);
+        Assertions.assertEquals(actual.get().getRoleName(), role.getRoleName());
```

Comment:
When using Optional, it's better to use assertPresent() before calling get(). This will make the test more readable and prevent NoSuchElementException if the Optional is empty.